xbmc/dialogs/GUIDialogPlayEject.cpp

```diff
@@ -89,10 +106,33 @@ void CGUIDialogPlayEject::OnInitWindow()
 bool CGUIDialogPlayEject::ShowAndGetInput(const CFileItem & item,
   unsigned int uiAutoCloseTime /* = 0 */)
 {
-  // Make sure we're actually dealing with a Disc Stub
-  if (!item.IsDiscStub())
+  // Make sure we're actually dealing with a Stub
+  if (!item.IsEFileStub() && !item.IsDiscStub())
     return false;
 
+  CURL stubUrl = item.GetURL();
+  isDisc = item.IsDiscStub();
```

Comment:
Don't do this. ShowAndGetInput should not mutate global state. Maybe all of this belongs somewhere else than in the eject dialog?